app/models/user.rb

```diff
@@ -0,0 +1,32 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+require 'dm-migrations'
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+  has n, :peeps, :through => Resource
+
+  property :id, Serial
+  property :username, String
+  property :password_digest, Text
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+  validates_confirmation_of :password, message:"Password and confirmation password do not match"
+  validates_presence_of :password
+
+  def password=(password)
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(username, password)
```

Comment:
Would it be possible to rewrite this method to use a guard clause that raises an error if the password is wrong? It might make the control flow even clearer.